### ISSUE TITLE
[App]: Git state and repository not appearing on experiment

### ISSUE THREAD

**Issue by miraodasilva** [wandb/wandb#5096]
### Current Behavior

I cannot see the git state and repository in my experiments. Please see below:
![a](https://user-images.githubusercontent.com/17463598/223538820-f137d7b0-2666-48a9-bea1-6f7a10fc7990.png)
However, a collaborator on the project has run the same command from the same folder, and in his experiments, git state and repository appear normally. Please see below:
![b](https://user-images.githubusercontent.com/17463598/223539043-c071268a-7242-4313-82b7-e16912eaf4fc.png)
We are both running the latest version of wandb, so I suppose either there is something wrong with my account or there is some kind of environment variable I'm not setting correctly? My github account is associated with my wandb account as well, so this should not be an issue.

Any help would be greatly appreciated :) 

### Expected Behavior

The expect behavior is explained above: the git state and git repository should appear normally.

### Environment

OS: Ubuntu

Browsers: Chrome

Version:

**Comment by billmdevs** [comment]
Hello @miraodasilva I don't precisely understand the issue in connection to wandb client. Are you using the personal github intergration as directed on wandb [here](https://docs.wandb.ai/guides/app/settings-page/user-settings#personal-github-integration)?

**Comment by miraodasilva** [comment]
Hi @billmdevs , apologies for the late response - yes I am using personal github integration, as described in the link you mentioned. However, unfortunately, I am still experiencing this issue.

**Comment by billmdevs** [comment]
Hi @miraodasilva thank you for getting back to me. Let me take a look and get back to you.

**Comment by miraodasilva** [comment]
I suppose I should ask: is there anything I should do in my wandb.init or anywhere else to log the git state/repository, other than integrating my github with wandb, which I have already done? Thanks a lot in advance.

**Comment by miraodasilva** [comment]
I wandb.init, I am setting the following parameters: 
- name: ${experiment_name}
- save_dir: ${experiment_dir}
- project: ${project_name}
- mode: online

and also am logged into my account as evidenced by typing "wandb login" in the terminal, which shows my account.

**Comment by miraodasilva** [comment]
Ok, I managed to identify and solve the bug. 

In short, it's due to Hydra (https://hydra.cc/docs/intro/) changing the working directory. 

When wandb tries to find the repo by looking at the current working directory via os.getcwd() (wandb/sdk/lib/git.py, class GitRepo), this has been changed by hydra and does not correspond to the directory that contains the actual git repo. So it is not found, and therefore not logged. 

Bit of a unique interaction but a lot of my colleagues were having the same issue, so might be worth pointing out (even though, effectively, the issue is with Hydra and not Wandb). Hope that helps, and thank you so much for the hard work on wandb. I think it's really the best visualization tool out there by miles!
